### ISSUE TITLE
musl is no longer thread-safe

### ISSUE THREAD

**Issue by sleirsgoevy** [OpenOrbis/OpenOrbis-PS4-Toolchain#94]
musl libc uses an internal thread counter (`libc.threads_minus_1`) to determine whether the program is multithreaded and locking is required. For programs that use Sony's pthread implementation directly, this counter is never incremented, and some operations (most importantly malloc/free) run unlocked, which they shouldn't do.

**Comment by Cryptogenic** [comment]
I'll have to think on an elegant solution to this problem, because the way fallback works is the implementations for the pthread functions are essentially stripped out and left for the dynamic linker to use the libkernel implementations. I don't see how these functions can easily be wrapped atm.

**Comment by Cryptogenic** [comment]
Ok so I came up with a few strategies and did some performance tests. The two main strategies I was considering was:

1. Setting the thread count to a non-zero value to force locking (easiest method, thought there may be too much performance overhead)
2. Writing wrappers (harder, also won't help if people directly invoke scePthread* functions)

After performance testing the first method, the performance impact was actually not even noticeable. Functions like `malloc()` took 0.000001 seconds with or without locking. Due to the performance overhead being so negligible + the fact that wrappers are an incomplete solution, I've decided to set the thread count to one in the CRT stub to just ensure locking occurs to force thread-safety,
